x/builder/ante/ante.go

```diff
@@ -93,10 +93,28 @@ func (ad BuilderDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool,
 			}
 		}
 
-		if err := ad.builderKeeper.ValidateBidInfo(ctx, topBid, bidInfo); err != nil {
+		if err := bd.builderKeeper.ValidateBidInfo(ctx, topBid, bidInfo); err != nil {
 			return ctx, errors.Wrap(err, "failed to validate auction bid")
 		}
 	}
 
 	return next(ctx, tx, simulate)
 }
+
+// ValidateTimeout validates that the timeout is greater than or equal to the expected block height
+// the bid transaction will be executed in.
+func (bd BuilderDecorator) ValidateTimeout(ctx sdk.Context, timeout int64) error {
+	currentBlockHeight := ctx.BlockHeight()
+
+	// If the mode is CheckTx or ReCheckTx, we increment the current block height by one to
+	// account for the fact that the transaction will be executed in the next block.
+	if ctx.IsCheckTx() || ctx.IsReCheckTx() {
+		currentBlockHeight++
+	}
```

Comment:
This is what I briefly talked about earlier today. Take the following scenario:
1. It is height H
2. searcher broadcasts with timeout H.

With current flow, we would accept the bid as timeout < H even though we know that when finalize block is executed, the bid would be rejected since the height would be updated to H + 1. Small fix to a potential grieving vector given timeouts.

Reply:
Resolving this per convo at stand up